pkg/cover/report.go

```diff
@@ -70,20 +78,26 @@ func MakeReportGenerator(target *targets.Target, kernelObject, srcDir, buildDir
 
 type file struct {
 	lines       map[int]line
+	functions   map[string]*function
 	totalPCs    map[uint64]bool
 	coverPCs    map[uint64]bool
 	totalInline map[int]bool
 	coverInline map[int]bool
 }
 
+type function struct {
+	totalPCs map[uint64]bool
+	coverPCs map[uint64]bool
+}
+
 type line struct {
 	count         map[int]bool
 	prog          int
 	uncovered     bool
 	symbolCovered bool
 }
 
-func (rg *ReportGenerator) Do(w io.Writer, progs []Prog) error {
+func (rg *ReportGenerator) Do(htmlBuf, csvBuf io.Writer, progs []Prog) error {
```

Comment:
I would assume caller or this will need only 1 format, no both. A better interface would be separate DoHTML and DoCVS.

Reply:
Addressed all your comments, please re-check.